lib/imgproxy.ex

```diff
@@ -154,18 +184,36 @@ defimpl String.Chars, for: Imgproxy do
   end
 
   #  @spec build_path(img_url :: String.t(), opts :: image_opts) :: String.t()
-  defp build_path(%Imgproxy{source_url: source_url, options: opts, extension: ext}) do
+  defp build_path(%Imgproxy{options: opts} = img) do
     ["/" | Enum.map(opts, &option_to_string/1)]
     |> Path.join()
-    |> Path.join(encode_source_url(source_url, ext))
+    |> Path.join(prepare_source_url(img))
+  end
+
+  defp prepare_source_url(img) do
+    img.source_url
+    |> optionally_encode_source_url(img)
+    |> optionally_add_extension(img)
   end
 
-  defp encode_source_url(source_url, nil) do
+  defp optionally_encode_source_url(source_url, %Imgproxy{source_url_encoding: :base64}) do
     Base.url_encode64(source_url, padding: false)
   end
 
-  defp encode_source_url(source_url, extension) do
-    encode_source_url(source_url, nil) <> "." <> extension
+  defp optionally_encode_source_url(source_url, _img) do
+    Path.join("plain", source_url)
```

Comment:
I believe in this case `source_url` may still need to be percent encoded based on the info block [here](https://docs.imgproxy.net/usage/processing#plain).

Reply:
I missed that! Update pushed.